plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePropertiesTable.java

```diff
@@ -0,0 +1,99 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.plugin.deltalake.transactionlog.MetadataEntry;
+import io.trino.plugin.deltalake.transactionlog.ProtocolEntry;
+import io.trino.plugin.deltalake.util.PageListBuilder;
+import io.trino.spi.Page;
+import io.trino.spi.connector.ColumnMetadata;
+import io.trino.spi.connector.ConnectorPageSource;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.FixedPageSource;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.SystemTable;
+import io.trino.spi.predicate.TupleDomain;
+
+import java.util.List;
+
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class DeltaLakePropertiesTable
+        implements SystemTable
+{
+    private static final String MIN_READER_VERSION_KEY = "delta.minReaderVersion";
+    private static final String MIN_WRITER_VERSION_KEY = "delta.minWriterVersion";
+
+    private final ConnectorTableMetadata tableMetadata;
+    private final MetadataEntry metadataEntry;
+    private final ProtocolEntry protocolEntry;
+
+    public DeltaLakePropertiesTable(SchemaTableName tableName, MetadataEntry metadataEntry, ProtocolEntry protocolEntry)
+    {
+        this.metadataEntry = requireNonNull(metadataEntry, "metadataEntry is null");
+        this.protocolEntry = requireNonNull(protocolEntry, "protocolEntry is null");
+
+        this.tableMetadata = new ConnectorTableMetadata(requireNonNull(tableName, "tableName is null"),
+                ImmutableList.<ColumnMetadata>builder()
+                        .add(new ColumnMetadata("key", VARCHAR))
+                        .add(new ColumnMetadata("value", VARCHAR))
+                        .build());
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        return new FixedPageSource(buildPages(tableMetadata, metadataEntry, protocolEntry));
+    }
+
+    private static List<Page> buildPages(ConnectorTableMetadata tableMetadata, MetadataEntry metadataEntry, ProtocolEntry protocolEntry)
```

Comment:
Feature names property (e.g. `delta.feature.columnMapping`) is also missing in the current implementation.

Reply:
Please take a look at:
```sql
CREATE TABLE test (c1 int) USING delta location 's3://starburstdata-test/test' TBLPROPERTIES ('delta.minReaderVersion'='3', 'delta.minWriterVersion'='7', 'delta.columnMapping.mode'='id');

SHOW TBLPROPERTIES test;
delta.columnMapping.maxColumnId	1
delta.columnMapping.mode	id
delta.feature.columnMapping	supported
delta.minReaderVersion	3
delta.minWriterVersion	7
Time taken: 0.17 seconds, Fetched 5 row(s)
```

Reply:
Those features were introduced in reader version 3 and writer version 7.

Reply:
Column mapping mode is supposed to have been present since reader version 2 and writer version 5. Perhaps the way writers write protocol entries have changed between writer versions in the Spark implementation.

I'll update the logic to include enabled features. It does not look like Databricks differentiates between features which are enabled only as `writerFeatures`, and features which are enabled as `readerFeatures`. I'll update the logic in `buildPages` to emit a `('delta.feature.<featureName>', 'supported')` row for every features in the union of `protocolEntry.readerFeatures` and `protocolEntry.writerFeatures`.

Reply:
> Column mapping mode is supposed to have been present since reader version 2 and writer version 5.

I knew. I meant "feature" concepts were introduced in the above versions.